frontend/src/components/features/chat/event-message.tsx

```diff
@@ -39,6 +42,53 @@ export function EventMessage({
   const shouldShowConfirmationButtons =
     isLastMessage && event.source === "agent" && isAwaitingUserConfirmation;
 
+  const { send, parsedEvents } = useWsClient();
+
+  // Check if there's already a UserFeedbackAction in the event stream
+  const hasFeedbackBeenSubmitted = React.useMemo(() => {
+    if (!parsedEvents) return false;
+
+    // If this is a finish action, check if there's a user_feedback action after it
+    if (isFinishAction(event)) {
+      const currentEventIndex = parsedEvents.findIndex(
+        (e) => e.id === event.id,
+      );
+      if (currentEventIndex === -1) return false;
+
+      // Check if there's a user_feedback action after this finish action
```

Comment:
@Openhands We should add a field in `UserFeedbackAction` to track which "event ID" this user feedback corresponds to -- and we should just check if a UserFeedbackAction with the current event ID exists